File: cpp/strings/1545-find-kth-bit-in-nth-binary-string.cpp
char findKthBit (int n, int k) {
    int mid = 1 << (n-1);

    if (n == 1) {
        return '0';
    }

    if (k < mid) {
        return findKthBit(n - 1, k);
    } else if (k > mid) {
        return findKthBit(n - 1, 2 * mid - k) == '0' ? '1' : '0';
    }

    return '1';
}

File: js/strings/1545-find-kth-bit-in-nth-binary-string.js
var findKthBit = function(n, k) {
    const S = (n) => {
        if (n === 1) {
            return '0';
        }
        const str = S(n - 1);
        return str + '1' + str.split('').map(char => char === '0' ? '1' : '0').reverse().join('');
    }
    
    return S(n)[k - 1];
};